designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/activities/ScenarioActivityRepository.scala

```diff
@@ -0,0 +1,81 @@
+package pl.touk.nussknacker.ui.process.repository.activities
+
+import db.util.DBIOActionInstances.DB
+import pl.touk.nussknacker.engine.api.deployment.{ScenarioActivity, ScenarioActivityId}
+import pl.touk.nussknacker.engine.api.process.{ProcessId, VersionId}
+import pl.touk.nussknacker.ui.api.description.scenarioActivity.Dtos.Legacy
+import pl.touk.nussknacker.ui.db.entity.AttachmentEntityData
+import pl.touk.nussknacker.ui.process.ScenarioAttachmentService.AttachmentToAdd
+import pl.touk.nussknacker.ui.process.repository.activities.ScenarioActivityRepository.ModifyCommentError
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+trait ScenarioActivityRepository {
```

Comment:
1. I see that you directly use the repository to record activity. I already do the service (`newactivity.ActivityService`) which was designed to handle it. We try to follow three layer architecture, IMO services shouldn't call their own repositories directly (they should call services). WDYT?
2. When I see multiple similar addXYZ methods, I have a feeling that it would be better to use the command pattern. Thanks to that we can extract a parent class for similar activities and have some common handlers for similar things. Also, it is easier to distinguish between a next adding method and totally different method like `findActivities`

Reply:
Ad 1 
I totally agree with 3-layer architecture, but would like to do it in next PR that I've started working on today. In next PR I intended to add handling of scenario activities, that were not persisted before (and were not added in this PR because of huge scope), like incoming and outgoing migrations, or technical updates. So:
- in this PR I replaced old ProcessActions with ScenarioActivities, but in some places I tried to not make too huge changes
- in next PR I would refactor it, including the service layer in between
 
 Would it be ok?
 
Ad 2 
Did you mean command pattern on repository level, or only service level? I would prefer to do it in next PR too, because it is related to the refactor from Ad 1

So I agree on both points, but would like to do this refactor in next PR I already started working on.

Reply:
The command pattern is valuable in any layer - everwhere where there are multiple similar methods. I see that you already did something in this direction in this PR as well. With regards to moving mentioned reactors to the next PR - it is fine for me :+1: